watch/watch.c

```diff
@@ -146,45 +145,51 @@ int main(int argc, char *argv[]) {
 		commandv[c] = argv[c];
 	}
 
-	/* 
+	/*
 	 * Initialize curses terminal with colours to be used.
 	 * Get terminal size too, we're going to need it.
 	 */
 	newterm(getenv("TERM"), stdout, stdin);
 	start_color();
 	init_pair(1, COLOR_BLACK, COLOR_WHITE);
 
+	/*
+	 * Declaring the left side of the information header, which contains
+	 * "Every χ second(s): ...", outside the loop because it's immutable
+	 * after we got the program we're going to run and the amount of time.
+	 */
+	static int left_len = 0;
+	static char left[256];
```

Comment:
Why `static`?
It would be good to set `left[0] = '\0'` in case of any faulty implementation of the displaying

Reply:
> Why static?

If I'm not erred, ``static`` variables are saved into the data segment, not in the stack one. This would be good in this case, where the variable content will not be changing anytime after it's declared.

> It would be good to set left[0] = '\0' in case of any faulty implementation of the displaying

Sure, good catch too.

Reply:
> > Why static?
> 
> If I'm not erred, `static` variables are saved into the data segment, not in the stack one. This would be good in this case, where the variable content will not be changing anytime after it's declared.

I don't think that's true considering that the pattern of creating a buffer is really common and it's not common at all to see the use of `static`. I think it would be better if there's any difference in the compiled assembly.

Reply:
> I don't think that's true considering that the pattern of creating a buffer is really common and it's not common at all to see the use of static.

Well, I read this in a website, not even my C book talks about it.
After I correct some things you pointed out, could you compare the two assembly files?

Reply:
Ok, but can you link the website? Usually `static` only makes sense inside functions/procedures that run several times, main will run once.

Reply:
I know some C, but I usually do research beyond my book and my mind to know if (and how) I am erred about something.
This time, I researched about ``static``, ended up on this response at StackOverflow:
https://stackoverflow.com/a/572550
It directly cites Wikipedia, but this affirmation is also found paraphrased at a GeeksforGeeks article. Going for Wikipedia, it doesn't actually cite any external sources --- not even an blog post, article or even "alchemy" of who wrote ---, but it's here for reference: https://en.wikipedia.org/wiki/Static_variable#Addressing

Reply:
I think it's better to resort to the more widely used pattern when in doubt

Reply:
> I think it's better to resort to the more widely used pattern when in doubt

I thought of using ``register``, but it would be out of place since it is meant for accessing and modifying the content of a variable faster.

So remove ``static`` or compare its assembly first?

Reply:
Remove `static` and compare its assembly later, maybe after merging. It would be better to contact someone more experienced with the language or try to find more resources about it.